src/azure-cli/azure/cli/command_modules/backup/_params.py

```diff
@@ -71,6 +72,11 @@
 target_tier_type = CLIArgumentType(help=target_tier_help, arg_type=get_enum_type(allowed_target_tier_type_chk_archivable), options_list=['--target-tier'])
 tier_type = CLIArgumentType(help=tier_help, arg_type=get_enum_type(allowed_tier_type), options_list=['--tier'])
 rehyd_priority_type = CLIArgumentType(help=rehyd_priority_type_help, arg_type=get_enum_type(allowed_rehyd_priority_type), options_list=['--rehydration-priority'])
+user_assigned_identity_id_type = CLIArgumentType(options_list=['--identity-id'], help="UserAssigned Identity Id to be used for CMK encryption, this will be applicable for encryption using userassigned identity")
+encryption_key_id_type = CLIArgumentType(options_list=['--encryption-key-id'], help="The encryption key id you want to use for encryption")
+infrastructure_encryption_type = CLIArgumentType(options_list=['--infrastructure-encryption'], arg_type=get_enum_type(['Enabled', 'Disabled']), help=infrastructure_encryption_type_help)
+user_assigned_type = CLIArgumentType(nargs='+', options_list=['--user-assigned'], help="Space-separated list of userassigned identities.")
+user_assigned_remove_type = CLIArgumentType(nargs='*', options_list=['--user-assigned'], help="Space-separated list of userassigned identities.")
```

Comment:
Can `user_assigned_type` be replaced by `user_assigned_remove_type`, so it's redundant?

Reply:
@zhoxing-ms: we want to keep nargs different - is there a better way to do this ?

Reply:
Could we not add `nargs` to `CLIArgumentType`, but declare it separately for different commands?

Reply:
Such as:
```Python
with self.argument_context('backup vault identity assign') as c:
    c.argument('user_assigned', nargs='+', user_assigned_type)

with self.argument_context('backup vault identity remove') as c:
    c.argument('user_assigned',nargs='*',  user_assigned_type)
```

Reply:
as discussed - keeping it as it is because we no longer have same  user_assigned_type help for both params